app/index.js

```diff
@@ -504,6 +504,14 @@ JhipsterGenerator.prototype.app = function app() {
     this.copy('gitignore', '.gitignore');
     this.copy('gitattributes', '.gitattributes');
 
+    // Create docker-compose file
+    if (this.devDatabaseType != "h2Memory" && this.devDatabaseType != "oracle") {
```

Comment:
I would add the docker-compose file also in case using h2 as dev database. You can you it to run and test the app in production profile without installing a the production database (mysql, psql...)